### PR TITLE
Azure CLI based DeploymentScript - Importing ACR Images

### PR DESCRIPTION
**Pull request by Gordonby**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Description

* Demonstrates how to use DeploymentScript to import public images to ACR
* Shows using an inline AzureCLI DeploymentScript resource (not PowerShell 😄)

## Changelog

*
*

### CODE REVIEW

File: quickstarts/microsoft.resources/deployment-script-azcli-acr-import/main.bicep [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+resource acr 'Microsoft.ContainerRegistry/registries@2021-12-01-preview' = {
+  name: AcrName
+  location: location
+  sku: {
+    name: 'Basic'
+  }
+}
+
+param managedIdName string = 'id-ContainerRegistryImport'
```
Comment (bmoore-msft) [inline quickstarts/microsoft.resources/deployment-script-azcli-acr-import/main.bicep:32]: move this up to the param section at the top

File: quickstarts/microsoft.resources/deployment-script-azcli-acr-import/main.bicep [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+resource depScriptId 'Microsoft.ManagedIdentity/userAssignedIdentities@2018-11-30' = {
+  name: managedIdName
+  location: location
+}
+
+resource rbac 'Microsoft.Authorization/roleAssignments@2020-08-01-preview' = [for roleDefId in RbacRolesNeeded: {
+  name: guid(roleDefId, depScriptId.id)
```
Comment (bmoore-msft) [inline quickstarts/microsoft.resources/deployment-script-azcli-acr-import/main.bicep:39]: add ```acr.id``` to the guid function params

File: quickstarts/microsoft.resources/deployment-script-azcli-acr-import/main.bicep [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    principalType: 'ServicePrincipal'
+  }
+}]
+
+@description('An array of fully qualified images names to import')
+param images array = [
+  'docker.io/bitnami/external-dns:latest'
+]
+
+//@batchSize(1)
```
Comment (bmoore-msft) [inline quickstarts/microsoft.resources/deployment-script-azcli-acr-import/main.bicep:53]: remove comment

File: quickstarts/microsoft.resources/deployment-script-azcli-acr-import/main.bicep [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+param date string = utcNow()
+
+@description('Version of the Azure CLI to use')
+param azCliVersion string = '2.30.0'
+
+@description('Deployment Script timeout')
+param timeout string = 'PT30M'
+
+@description('The retention period for the deployment script')
+param retention string = 'P1D'
```
Comment (bmoore-msft) [inline quickstarts/microsoft.resources/deployment-script-azcli-acr-import/main.bicep:17]: remove this and just use a literal value

File: quickstarts/microsoft.resources/deployment-script-azcli-acr-import/main.bicep [hunk 5]
```diff
@@ -0,0 +1,91 @@
+@description('The name of the Azure Container Registry')
+param AcrName string
```
Comment (bmoore-msft) [inline quickstarts/microsoft.resources/deployment-script-azcli-acr-import/main.bicep:2]: add a defaultValue with ```uniqueString```

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: quickstarts/microsoft.resources/deployment-script-azcli-acr-import/main.bicep [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+param azCliVersion string = '2.30.0'
+
+@description('Deployment Script timeout')
+param timeout string = 'PT30M'
+
+@description('The retention period for the deployment script')
+param retention string = 'P1D'
+
+@description('An array of Azure RoleId that are required for the DeploymentScript resource')
+param RbacRolesNeeded array = [
```
[the remark this answers is not in the document]
Comment (Gordonby) [reply quickstarts/microsoft.resources/deployment-script-azcli-acr-import/main.bicep:20]: I like the pattern of providing an array of rbacRoles needed. It doesn't add a great deal to this particular sample, but it makes it easily extensible.

[the remark this answers is not in the document]
Comment (Gordonby) [reply quickstarts/microsoft.resources/deployment-script-azcli-acr-import/main.bicep:11]: Happy to use a var if this is a real problem though 😺

File: quickstarts/microsoft.resources/deployment-script-azcli-acr-import/main.bicep [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+  }
+}]
+
+@description('An array of fully qualified images names to import')
+param images array = [
+  'docker.io/bitnami/external-dns:latest'
+]
+
+//@batchSize(1)
+resource createAddCertificate 'Microsoft.Resources/deploymentScripts@2020-10-01' = [for image in images: {
```
[the remark this answers is not in the document]
Comment (Gordonby) [reply quickstarts/microsoft.resources/deployment-script-azcli-acr-import/main.bicep]: Resolving as this sample has pivoted to use the module from the bicep registry

File: quickstarts/microsoft.resources/deployment-script-azcli-acr-import/main.bicep (hunk 6 above)
[the remark this answers is not in the document]
Comment (Gordonby) [reply quickstarts/microsoft.resources/deployment-script-azcli-acr-import/main.bicep]: Resolving as this sample has pivoted to use the module from the bicep registry

**Comment by Gordonby** [comment]
Pivoted this sample to leverage the module from the Public Bicep Registry
https://github.com/Azure/bicep-registry-modules/pull/65

File: quickstarts/microsoft.resources/deployment-script-azcli-acr-import/README.md [hunk 8]
```diff
@@ -17,8 +17,9 @@
 
 ## Sample overview
 
-This template uses the DeploymentScript resource to create import a number of Container Images from public sources to an Azure Container Registry. A new Azure Container Registry is created.
+This template leverages the Import ACR module from the bicep registry to import public container images into an Azure Container Registry.. A new Azure Container Registry is created.
```
Comment (bmoore-msft) [inline quickstarts/microsoft.resources/deployment-script-azcli-acr-import/README.md:20]: remove extra period

File: quickstarts/microsoft.resources/deployment-script-azcli-acr-import/README.md [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+
+See the [Import-ACR] module in the Bicep Registry for more information.
+See the [docs](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/deployment-script-template?tabs=CLI) for more information on the deployment script resource.
+
+## Deployment steps
+
+You can click the "deploy to Azure" button at the beginning of this document or follow the instructions for command line deployment using the scripts in the root of this repo.
+## Notes
+
+The AzureCLI is used in preference to Azure PowerShell primarily to provide a sample of using the Azure CLI DeploymentScripts.
```
Comment (bmoore-msft) [inline quickstarts/microsoft.resources/deployment-script-azcli-acr-import/README.md:30]: remove this paragraph

File: quickstarts/microsoft.resources/deployment-script-azcli-acr-import/README.md [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+[![Deploy To Azure US Gov](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/deploytoazuregov.svg?sanitize=true)](https://portal.azure.us/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2Fquickstarts%2Fmicrosoft.resources%2Fdeployment-script-azcli-acr-import%2Fazuredeploy.json)
+[![Visualize](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/visualizebutton.svg?sanitize=true)](http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2Fquickstarts%2Fmicrosoft.resources%2Fdeployment-script-azcli-acr-import%2Fazuredeploy.json)
+
+## Sample overview
+
+This template leverages the Import ACR module from the bicep registry to import public container images into an Azure Container Registry.. A new Azure Container Registry is created.
+
+See the [Import-ACR] module in the Bicep Registry for more information.
```
Comment (bmoore-msft) [inline quickstarts/microsoft.resources/deployment-script-azcli-acr-import/README.md:22]: add a link to the module

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

**Pull request merged**